module_analysis/2_add_annotations.py

```diff
@@ -0,0 +1,246 @@
+#     language: python
+#     name: conda-env-core_acc-py
+# ---
+
+# # Add annotations
+#
+# This notebook takes the dataframe with information about module composition and their labels and adds additional annotations including:
+#
+# 1. Which gene is contained within the modules (both gene id and gene name)
+# 2. KEGG pathways that genes are found in
+# 3. GO pathways genes are found in
+# 4. Regulon/operon genes are found in
+
+# +
+# %load_ext autoreload
+# %autoreload 2
+# %matplotlib inline
+import os
+import random
+import scipy
+import pandas as pd
+import numpy as np
+import seaborn as sns
+import matplotlib.pyplot as plt
+from core_acc_modules import utils, paths
+
+random.seed(1)
+# -
+
+# User param
+method = "affinity"
+
+# +
+# Import module labels
+pao1_module_label_filename = os.path.join(
+    paths.LOCAL_DATA_DIR, "pao1_gene_module_labels.tsv"
+)
+pa14_module_label_filename = os.path.join(
+    paths.LOCAL_DATA_DIR, "pa14_gene_module_labels.tsv"
+)
+
+pao1_module_labels = pd.read_csv(
+    pao1_module_label_filename, sep="\t", index_col=0, header=0
+)
+pa14_module_labels = pd.read_csv(
+    pa14_module_label_filename, sep="\t", index_col=0, header=0
+)
+
+# +
+# Import gene memberships
+pao1_membership_filename = os.path.join(
+    paths.LOCAL_DATA_DIR, f"pao1_modules_{method}.tsv"
+)
+pa14_membership_filename = os.path.join(
+    paths.LOCAL_DATA_DIR, f"pa14_modules_{method}.tsv"
+)
+
+pao1_membership = pd.read_csv(pao1_membership_filename, sep="\t", index_col=0, header=0)
+pa14_membership = pd.read_csv(pa14_membership_filename, sep="\t", index_col=0, header=0)
+
+# +
+# Import gene metadata
+pao1_gene_annot_filename = paths.GENE_PAO1_ANNOT
+pa14_gene_annot_filename = paths.GENE_PA14_ANNOT
+
+pao1_gene_annot = pd.read_csv(pao1_gene_annot_filename, index_col=0, header=0)
+pa14_gene_annot = pd.read_csv(pa14_gene_annot_filename, index_col=0, header=0)
+# -
+
+pao1_gene_annot = pao1_gene_annot["Name"].to_frame("gene name")
+pa14_gene_annot = pa14_gene_annot["Name"].to_frame("gene name")
+
+# ## Add module labels
+
+# Add module labels
+pao1_gene_module_labels = pao1_membership.merge(
+    pao1_module_labels, left_on="module id", right_index=True
+)
+pa14_gene_module_labels = pa14_membership.merge(
+    pa14_module_labels, left_on="module id", right_index=True
+)
+
+# ## Add gene names
+
+# Add gene names
+pao1_gene_module_labels = pao1_gene_module_labels.merge(
+    pao1_gene_annot, left_index=True, right_index=True
+)
+pa14_gene_module_labels = pa14_gene_module_labels.merge(
+    pa14_gene_annot, left_index=True, right_index=True
+)
+
+print(pao1_gene_module_labels.shape)
+pao1_gene_module_labels.head()
+
+print(pa14_gene_module_labels.shape)
+pa14_gene_module_labels.head()
+
+# ## Add KEGG pathways
+#
+# For each pathway, what genes are contained in it
+
+# +
+pao1_pathway_filename = "https://raw.githubusercontent.com/greenelab/adage/master/Node_interpretation/pseudomonas_KEGG_terms.txt"
```

Comment:
Might be better to link to a specific commit here (instead of `master` since it could change).